### PR TITLE
Add contribution instructions to stdlib docs

### PR DESCRIPTION
**Pull request by pirate**
Generally programming language docs have instructions on how to contribute changes.

I couldn't find any in the rust docs, so I figured I'd add an instructions section, let me know if this belongs somewhere else!

### CODE REVIEW

**Comment by steveklabnik** [comment]
This should probably link to https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md

**Comment by pirate** [comment]
@steveklabnik added a link to the guidelines, also mentioned the #rust IRC channel and changed some wording.

File: src/libstd/lib.rs [hunk 1]
```diff
@@ -94,6 +94,16 @@
...
 //!
+//! # Contributing changes to the documentation
+//!
+//! Check out the rust contribution guidelines [here](https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md).
+//! The source for this documentation can be found on [Github](https://github.com/rust-lang/rust/tree/master/src/libstd).
+//! To contribute changes, make sure you read the guidelines first, then submit
+//! pull-requests for your suggested changes.
+//! 
+//! Contributions are appreciated! If you see a part of the docs that can be
+//! improved, submit a PR, or chat with us first on irc.mozilla.org #rust.
```
Comment (aturon) [inline src/libstd/lib.rs]: Might want to suggest #rust-docs instead?

File: src/libstd/lib.rs [hunk 2]
```diff
@@ -94,6 +94,16 @@
 //! compiler - but they are documented here the same). Like the prelude, the
 //! standard macros are imported by default into all crates.
 //!
+//! # Contributing changes to the documentation
+//!
+//! Check out the rust contribution guidelines [here](https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md).
+//! The source for this documentation can be found on [Github](https://github.com/rust-lang/rust/tree/master/src/libstd).
```
Comment (aturon) [inline src/libstd/lib.rs]: A bunch of the source lives in `libcore`, `liballoc`, `libcollections`, should probably include pointers there as well.

**Review by aturon — COMMENTED** [review]
Just a couple of nits.

File: src/libstd/lib.rs (hunk 2 above)
Comment (pirate) [inline src/libstd/lib.rs]: Do we want to link to all the various source repos, or shall I just link to https://github.com/rust-lang instead?

Comment (steveklabnik) [inline src/libstd/lib.rs]: I think just linking to the main repo is a good idea.

**Comment by alexcrichton** [comment]
bors: r=aturon

**Pull request merged**